src/V3Error.h

```diff
@@ -246,33 +247,52 @@ inline std::ostream& operator<<(std::ostream& os, const V3ErrorCode& rhs) {
     return os << rhs.ascii();
 }
 
-// ######################################################################
-
-class V3Error final {
-    // Base class for any object that wants debugging and error reporting
-
+class V3ErrorGuarded VL_NOT_FINAL {
     using MessagesSet = std::set<std::string>;
+
+protected:
     using ErrorExitCb = void (*)(void);
 
-private:
-    static bool s_describedWarnings;  // Told user how to disable warns
-    static bool s_describedWeb;  // Told user to see web
+    V3ErrorGuarded() {
+        std::cerr << ("Static class");
+        std::abort();
+    }
+
+    static VerilatedMutex s_mutex;  // Make sure only single thread is in class
+
+    static bool s_describedWarnings VL_GUARDED_BY(s_mutex);  // Told user how to disable warns
     static std::array<bool, V3ErrorCode::_ENUM_MAX>
-        s_describedEachWarn;  // Told user specifics about this warning
+        s_describedEachWarn VL_GUARDED_BY(s_mutex);  // Told user specifics about this warning
     static std::array<bool, V3ErrorCode::_ENUM_MAX>
-        s_pretendError;  // Pretend this warning is an error
+        s_pretendError VL_GUARDED_BY(s_mutex);  // Pretend this warning is an error
+    static int s_tellManual
+        VL_GUARDED_BY(s_mutex);  // Tell user to see manual, 0=not yet, 1=doit, 2=disable
+    static V3ErrorCode s_errorCode VL_GUARDED_BY(s_mutex);  // Error string being formed will abort
+    static bool
+        s_errorSuppressed VL_GUARDED_BY(s_mutex);  // Error being formed should be suppressed
+    static MessagesSet s_messages VL_GUARDED_BY(s_mutex);  // What errors we've outputted
+    static ErrorExitCb
+        s_errorExitCb VL_GUARDED_BY(s_mutex);  // Callback when error occurs for dumping
+
+    static string msgPrefixNoLock() VL_REQUIRES(s_mutex) VL_MT_SAFE;  // returns %Error/%Warn
+    static string warnMoreNoLock() VL_REQUIRES(s_mutex) VL_MT_SAFE;
+    static void errorExitCbNoLock(ErrorExitCb cb) VL_REQUIRES(s_mutex) { s_errorExitCb = cb; }
+    static bool isError(V3ErrorCode code, bool supp) VL_REQUIRES(s_mutex) VL_MT_SAFE;
+    static void vlAbortOrExit() VL_REQUIRES(s_mutex) VL_MT_SAFE;
+};
+
+// ######################################################################
+class V3Error final : protected V3ErrorGuarded {
```

Comment:
```suggestion

class V3Error final : protected V3ErrorGuarded {
```

An is-a relationship might not be ideal here, as it makes it easy to mistakenly call a unguarded function.  What does a has-a relationship with guarded look like instead?